examples/siro_sandbox/server/keyframe_utils.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Meta Platforms, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Any
+
+
+def update_consolidated_keyframe(consolidated_keyframe, inc_keyframe):
+    """
+    A "consolidated" keyframe is several incremental keyframes merged together.
+    See nearly duplicate logic in habitat-sim Recorder::addLoadsCreationsDeletions.
+    Note the simplification logic here: if an instance is created and then
+    later deleted, the latest consolidated keyframe won't have either the
+    creation, update, or deletion.
+    """
+    assert consolidated_keyframe is not None
+    assert inc_keyframe is not None
+
+    def ensure_list(keyframe, key):
+        if key not in keyframe:
+            keyframe[key] = []
+
+    def ensure_dict(keyframe, key):
+        if key not in keyframe:
+            keyframe[key] = {}
+
+    # append loads
+    if "loads" in inc_keyframe:
+        ensure_list(consolidated_keyframe, "loads")
+        consolidated_keyframe["loads"] += inc_keyframe["loads"]
+
+    # add or update stateUpdates based on instanceKey
+    if "stateUpdates" in inc_keyframe:
+        ensure_list(consolidated_keyframe, "stateUpdates")
+        for state_update in inc_keyframe["stateUpdates"]:
+            key = state_update["instanceKey"]
+            state = state_update["state"]
+            found = False
+            for con_state_update in consolidated_keyframe["stateUpdates"]:
+                if con_state_update["instanceKey"] == key:
+                    con_state_update["state"] = state
+                    found = True
+            if not found:
+                consolidated_keyframe["stateUpdates"].append(state_update)
+
+    # if "rigUpdates" in inc_keyframe:
```

Comment:
Shall we remove this?

Reply:
I don't feel strongly about these commented blocks.

They will eventually be uncommented to allow skinned meshes to be transmitted to remote clients.